packages/manager/apps/web/client/app/hosting/database/hosting-database.service.js

```diff
@@ -27,9 +37,144 @@ angular.module('services').service(
       this.OvhHttp = OvhHttp;
       this.Poller = Poller;
       this.OvhApiOrderCatalogPublic = OvhApiOrderCatalogPublic;
+      this.WucOrderCartService = WucOrderCartService;
       this.iceberg = iceberg;
     }
 
+    static getPrivateSqlCategory(catalog) {
+      const offers = catalog.addons
+        .filter(({ planCode }) =>
+          planCode.startsWith(PRIVATE_SQL_PLAN_CODE_PREFIX),
+        )
+        .map((dbGroup) => ({
+          ...dbGroup,
+          productSize: dbGroup.product.split('-')[2],
+          tracking: DB_OFFERS.PRIVATE.TRACKING,
+        }));
+      // init db engines
+      offers.forEach((dbOffer) => {
+        const dbms = dbOffer.configurations
+          .find(({ name }) => name === 'engine')
+          .values.map((db) => {
+            const [dbName, dbVersion] = db.split('_');
+            return { db, dbName, dbVersion };
+          });
+        const groupedDbms = groupBy(dbms, 'dbName');
+        const engines = Object.keys(groupedDbms).map((dbGroup) => ({
+          dbGroup,
+          versions: groupedDbms[dbGroup],
+        }));
+
+        set(dbOffer, 'engines', engines);
+      });
```

Comment:
Can you use reduce here please, it would be more performant (1 loop instead of 3) and also allow you not to use lodash set)